### PR TITLE
Travis OSX

### PR DESCRIPTION
**Pull request by rijobro**
Get OSX to compile on Travis.
- Use absolute paths
- Give full paths to python executable, library and include directory for either python2 or python3.\
- Upgrade to boost 1.66

### CODE REVIEW

**Comment by KrisThielemans** [comment]
failure related to `ismrmrd.dylib` again on only 1 of the jobs (which is quite weird). I guess need to check if the `mark_as_superbuild` trick worked. Could `cat` the `SIRF-prefix/..../SIRF-build/CMakeCache.txt` I guess. Presumably should have added something to the `CMAKE*FLAGS`

**Comment by rijobro** [comment]
changes made. there are currently 3 matrices, all of which build successfully.
1. python2, abs_paths
2. python3, abs_paths
3. python3, rpaths
So it looks like python3 doesn't need absolute paths, whereas python2 does.

**Comment by casperdcl** [comment]
closes #72

**Comment by casperdcl** [comment]
@KrisThielemans @rijobro I squashed all the commits (diff from master) into one commit (https://github.com/CCPPETMR/SIRF-SuperBuild/pull/80/commits/26a201b2552e6b32c3abe64dab6999412b6ac007) attributed to @rijobro, then synced `.travis.yml` with [SIRF#79 (osx-travis)](https://github.com/CCPPETMR/SIRF/pull/79). What needs reviewing is:

- README.md
- SuperBuild/External_STIR.cmake
- version_config.cmake

The rest lgtm

File: .travis.yml [hunk 1]
```diff
@@ -11,42 +11,57 @@ language: cpp
...
+   env: EXTRA_BUILD_FLAGS="-DUSE_ITK=ON -DUSE_SYSTEM_FFTW3=ON -DUSE_SYSTEM_HDF5=ON" CC=gcc-5 CXX=g++-5 PYMVER=3
+ - os: osx
+   python: 2.7
+   # +itk -hdf5 +swig
+   env: EXTRA_BUILD_FLAGS="-DUSE_ITK=ON -DUSE_SYSTEM_HDF5=OFF -DUSE_SYSTEM_SWIG=ON" CC=gcc CXX=g++ PYMVER=2
+
 env:
  global:
-  - BUILD_FLAGS="-DCMAKE_BUILD_TYPE=Release"
+  - BUILD_FLAGS="-DCMAKE_BUILD_TYPE=Release -DUSE_SYSTEM_Boost=ON"
```
Comment (KrisThielemans) [inline .travis.yml]: not sure why we want this for all builds. I'd prefer to have one on Linux and one on Mac with "our" Boost build. I'm not 100% sure if it'll work on Mac now though.

File: README.md [hunk 2]
```diff
@@ -5,46 +5,22 @@ needed to compile SIRF and Gadgetron, and automatically build SIRF and Gadgetron
 There is still a small number of libraries that are not installed
 by the SuperBuild, [see below for more info for your operating system](#os-specific-information).
 
-## Table of Contents
```
Comment (KrisThielemans) [inline README.md]: why are we removing this?

File: README.md [hunk 3]
```diff
@@ -5,46 +5,22 @@ needed to compile SIRF and Gadgetron, and automatically build SIRF and Gadgetron
...
-## Dependencies <a name="Dependencies"></a>
-
-The SuperBuild depends on CMake >= 3.7.
-
-If you are building Gadgetron there are a series of [additional dependencies](https://github.com/gadgetron/gadgetron/wiki/List-of-Dependencies), which must be met.
-
-## Generic instructions <a name="Generic_instructions"></a>
+## Dependencies
+
+The superBuild depends on CMake >= 3.7.
```
Comment (KrisThielemans) [inline README.md]: good to add, but we do need to point somewhere for Gadgetron dependencies. Those are now on our own Wiki pages as well.

File: README.md [hunk 4]
```diff
@@ -208,34 +178,20 @@ cmake ../SIRF-SuperBuild -DSIRF_TAG=<a valid hash>
...
+### Installation instructions for Docker
 
 They can be found [here](https://github.com/CCPPETMR/SIRF/wiki/SIRF-SuperBuild-on-Docker)
 
-## Advanced installation <a name="Advanced_installation"></a>
-
-### SIRF and MATLAB <a name="SIRF_and_MATLAB"></a>
-SIRF can be used from MATLAB. For more information on this, see our [SIRF and MATLAB page](SIRF-and-MATLAB).
-
-### Compiling against your own packages <a name="Compiling_own_packages"></a>
```
Comment (KrisThielemans) [inline README.md]: why is this removed?

File: version_config.cmake [hunk 5]
```diff
@@ -35,7 +35,7 @@ set(Armadillo_MD5 c601f3a5ec6d50666aa3a539fa20e6ca )
 if(WIN32)
   # Just use precompiled version
   # TODO would prefer the next zip file but for KT using an ftp URL times-out (firewall?)
-  set(FFTW3_URL https://www.ccppetmr.ac.uk/sites/www.ccppetmr.ac.uk/files/downloads/fftw-3.3.5-dll64.zip )
+  set(FFTW3_URL ftp://ftp.fftw.org/pub/fftw/fftw-3.3.5-dll64.zip )
```
Comment (KrisThielemans) [inline version_config.cmake]: this is another unwanted revert

File: SuperBuild/External_STIR.cmake [hunk 6]
```diff
@@ -54,6 +54,7 @@ if(NOT ( DEFINED "USE_SYSTEM_${externalProjName}" AND "${USE_SYSTEM_${externalPr
         -DCMAKE_INSTALL_PREFIX=${STIR_Install_Dir}
         -DGRAPHICS=None
         -DCMAKE_CXX_STANDARD=11
+        -DDISABLE_ITK=On
```
Comment (KrisThielemans) [inline SuperBuild/External_STIR.cmake]: unwanted revert

**Comment by casperdcl** [comment]
@KrisThielemans yes I thought it was strange too which is why I asked... there were quite a few merge commits & commits with 'rebase' commented which I guess was @rijobro having a little difficulty keeping in sync with `master`. I ultimately only squashed into a single commit (aka setting up a squash-merge, guaranteeing that I didn't change the effect of merging) because I couldn't rebase without loads of conflicts. I guess @rijobro you could

```
git checkout OSX_abs_path
git fetch
git reset --hard origin/OSX_abs_path
```
and then manually revert the bits of 26a201b which you don't recognise

File: .travis.yml (hunk 1 above)
Comment (casperdcl) [reply .travis.yml]: it was in all the builds (incl the mac build fixes) so I put it into the global one

**Comment by KrisThielemans** [comment]
yeah well. too scary for me. I'd rather create a new branch from master and fix only `.travis.yml` and `SuperBuild.cmake`. The rest should not change. (I'm not sure why `External_HDF5.cmake` was changed the way it was).

Seems that the `.travis.yml` on https://github.com/CCPPETMR/SIRF/pull/79 is almost what we need here as well.

File: .travis.yml (hunk 1 above)
Comment (KrisThielemans) [reply .travis.yml]: ok. I've created #84  to fix this later.

Comment (rijobro) [reply .travis.yml]: should `USE_SYSTEM_Boost=OFF` should work fine for Macs. Just put it in to speed up compilation whilst I was debugging.

File: README.md (hunk 2 above)
Comment (rijobro) [reply README.md]: Not sure what's happened here. Maybe I did something wrong with the merge because I didn't edit this file on this branch. I would ignore all changes to the `README.md` file (especially as it was me that updated it on the master).
